snips_nlu/slot_filler/feature_factory.py

```diff
@@ -408,9 +442,15 @@ def entities(self, value):
             self.args["entities"] = value
 
     def fit(self, dataset, intent):
-        self.entities = extract_intent_entities(
+        entities_names = extract_intent_entities(
             dataset, lambda e: not is_builtin_entity(e))[intent]
-        self.entities = list(self.entities)
+        extensible = self.entity_filter.get(AUTOMATICALLY_EXTENSIBLE)
+        if extensible is not None:
+            entities_names = [
+                e for e in entities_names
+                if dataset[ENTITIES][e][AUTOMATICALLY_EXTENSIBLE] == extensible
+            ]
```

Comment:
I'm not sure it can be done though since the `entity_filter` in the `extract_intent_entities` takes the entity name as argument while we need the entity itself to access its `automatically_extensible` attribute

Reply:
Ah indeed.

Reply:
I mean it could be done, by extracting the right list of dataset entity names depending on the filter and then passing it to the `extract_intent_entities` function but I don't find it particularly more elegant:
```python
allowed_entities = set(dataset[ENTITIES])
        extensible = self.entity_filter.get(AUTOMATICALLY_EXTENSIBLE)
        if extensible is not None:
            allowed_entities = set(
                e for e in allowed_entities
                if dataset[ENTITIES][e][AUTOMATICALLY_EXTENSIBLE] == extensible
            )
entities_names = extract_intent_entities(
     dataset,
     lambda e: not is_builtin_entity(e) and e in allowed_entities
     )[intent]
```